### PR TITLE
bugfix: fix various resource leaks

### PR DESCRIPTION
**Pull request by i-pankrat**
Fix memory and descriptor leaks.

### CODE REVIEW

**Review by glitsj16 — APPROVED** [review]
Looks fine. Thanks for the patches!

**Comment by i-pankrat** [comment]
Seems that there are some troubles. I'll double-check.

File: src/firemon/netstats.c [hunk 1]
```diff
@@ -154,6 +154,7 @@ static void print_proc(int index, int itv, int col) {
 		// the sandbox doesn't have a --net= option, don't print
 		if (cmd)
 			free(cmd);
+		free(name);
 		return;
 	}
```
Comment (kmk3) [inline src/firemon/netstats.c:159]:
```suggestion
		// the sandbox doesn't have a --net= option, don't print
		free(name);
		if (cmd)
			free(cmd);
		return;
	}
    free(name);
```

Free in reverse order and do `free(name)` early.

File: src/firemon/netstats.c [hunk 2]
```diff
@@ -189,7 +190,7 @@ static void print_proc(int index, int itv, int col) {
 		free(cmd);
 	if (user)
 		free(user);
-
+	free(name);
```
Comment (kmk3) [inline src/firemon/netstats.c:193]:
```suggestion
```

Undo; see above.

File: src/jailcheck/access.c [hunk 3]
```diff
@@ -80,10 +80,13 @@ void access_setup(const char *directory) {
 	FILE *fp = fopen(test_file, "w");
 	if (!fp) {
 		printf("Warning: I cannot create test file in directory %s, skipping...\n", directory);
+		free(path);
+		free(test_file);
 		return;
 	}
```
Comment (kmk3) [inline src/jailcheck/access.c:86]:
```suggestion
	if (!fp) {
		printf("Warning: I cannot create test file in directory %s, skipping...\n", directory);
		free(test_file);
		free(path);
		return;
	}
```

Free in reverse order.

File: src/jailcheck/noexec.c [hunk 4]
```diff
@@ -109,5 +110,6 @@ void noexec_test(const char *path) {
 	int status;
 	wait(&status);
 	int rv = unlink(fname);
+	free(fname);
 	(void) rv;
```
Comment (kmk3) [inline src/jailcheck/noexec.c:114]:
```suggestion
	int rv = unlink(fname);
	(void) rv;
	free(fname);
```

File: src/firejail/fs.c
Comment (kmk3) [inline src/firejail/fs.c:1]: I concur that the array itself should also be freed (I made related changes in
a WIP refactor branch awhile ago), though these changes break scan-build in CI
and they are more complex than the rest of the PR (which already changes many
parts of the codebase), so please remove the changes in this file for now.

File: src/profstats/main.c [hunk 5]
```diff
@@ -344,7 +344,7 @@ int main(int argc, char **argv) {
 		if (cnt_seccomp > (seccomp + 1))
 			cnt_seccomp = seccomp + 1;
 		if (cnt_restrict_namespaces > (restrict_namespaces + 1))
-			cnt_seccomp = restrict_namespaces + 1;
+			cnt_restrict_namespaces = restrict_namespaces + 1;
```
Comment (kmk3) [inline src/profstats/main.c:347]: This looks good but it's unrelated to everything else; please do it in a
separate PR.

**Review by kmk3 — CHANGES_REQUESTED** [review]
Looks good overall, with a few requested changes.

File: src/firecfg/desktop_files.c [hunk 6]
```diff
@@ -302,6 +302,7 @@ void fix_desktop_files(const char *homedir) {
 			printf("   %s skipped: file exists\n", filename);
 			if (change_exec)
 				free(change_exec);
+			free(outname);
 			continue;
```
[the remark this answers is not in the document]
Comment (i-pankrat) [reply src/firecfg/desktop_files.c]: Done.

File: src/firejail/bandwidth.c [hunk 7]
```diff
@@ -198,6 +198,7 @@ static void read_bandwidth_file(pid_t pid) {
 
 		fclose(fp);
 	}
+	free(fname);
 }
```
[the remark this answers is not in the document]
Comment (i-pankrat) [reply src/firejail/bandwidth.c]: Done

File: src/firemon/netstats.c (hunk 1 above)
Comment (i-pankrat) [reply src/firemon/netstats.c]: Done

File: src/firemon/netstats.c (hunk 2 above)
Comment (i-pankrat) [reply src/firemon/netstats.c]: Done

File: src/jailcheck/access.c (hunk 3 above)
Comment (i-pankrat) [reply src/jailcheck/access.c]: Done

File: src/jailcheck/noexec.c (hunk 4 above)
Comment (i-pankrat) [reply src/jailcheck/noexec.c]: Done

File: src/firejail/fs.c
Comment (i-pankrat) [reply src/firejail/fs.c:1]: Okay, do you need help with that?

File: src/profstats/main.c (hunk 5 above)
Comment (i-pankrat) [reply src/profstats/main.c]: Okay, see #6369

**Review by kmk3 — COMMENTED** [review]
Please amend the commit message title to match the PR title so that it's
consistent with the RELNOTES.

Also, it would be nice to have the PR description in the commit itself as well:

```
bugfix: fix various resource leaks

Fix memory and descriptor leaks.
```

File: src/firejail/fs.c
Comment (i-pankrat) [reply src/firejail/fs.c:1]: Yeah, you're right. It's correct, I didn't notice that the 'enames' members are stored in the 'noblacklist' array.

**Review by kmk3 — APPROVED** [review]
Nicely done; thanks for the many fixes!

**Pull request merged**